### ISSUE TITLE
Multiple HTML validity tests fail based on the culture settings

### ISSUE THREAD

**Issue by HellBrick** [AngleSharp/AngleSharp#482]
To reproduce the issue add the following class to `AngleSharp.Core.Tests`:
```C#
using System.Globalization;
using System.Threading;
using NUnit.Framework;

namespace AngleSharp.Core.Tests
{
	[SetUpFixture]
	public class Setup
	{
		[OneTimeSetUp]
		public void RunBeforeAnyTests() => Thread.CurrentThread.CurrentCulture = new CultureInfo( "ru" );
	}
}
```

It should cause the following 7 tests to fail:
![Failed tests](http://i.imgur.com/jengO0K.png)

With the same exception:
```
System.FormatException : Input string was not in a correct format.
   at System.Number.ParseDouble(String value, NumberStyles options, NumberFormatInfo numfmt)
   at AngleSharp.Html.InputTypes.BaseInputType.ToNumber(String value) in F:\Software\AngleSharp\src\AngleSharp\Html\InputTypes\BaseInputType.cs:line 333
   at AngleSharp.Html.InputTypes.NumberInputType.Check(IValidityState current) in F:\Software\AngleSharp\src\AngleSharp\Html\InputTypes\NumberInputType.cs:line 34
   at AngleSharp.Html.Dom.HtmlInputElement.Check(ValidityState state) in F:\Software\AngleSharp\src\AngleSharp\Html\Dom\Internal\HtmlInputElement.cs:line 370
   at AngleSharp.Core.Tests.Html.ValidityRangeOverflowTests.TestRangeoverflowInputNumber4() in F:\Software\AngleSharp\src\AngleSharp.Core.Tests\Html\ValidityRangeOverflow.cs:line 1367
```

Probably has something to do with the decimal symbol being `,` instead of `.`.

**Comment by FlorianRappl** [comment]
Good catch - will fix them. All of them are due to `Parse` being used without `CultureInfo.InvariantCulture`.

**Comment by FlorianRappl** [comment]
Fixed, see PR #484.
